pkg/shared/util/grpc_utils.go

```diff
@@ -20,18 +20,77 @@ import (
 	"context"
 	"fmt"
 	"log"
+	"strings"
 	"time"
 
+	"github.com/Masterminds/semver/v3"
 	"github.com/numaproj/numaflow-go/pkg/info"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/credentials/insecure"
 	"google.golang.org/grpc/status"
 
+	"github.com/numaproj/numaflow"
 	sdkerr "github.com/numaproj/numaflow/pkg/sdkclient/error"
 	resolver "github.com/numaproj/numaflow/pkg/sdkclient/grpc_resolver"
 )
 
+func checkConstraint(version *semver.Version, constraint string) error {
+	c, err := semver.NewConstraint(constraint)
+	if err != nil {
+		return fmt.Errorf("error parsing constraint: %w", err)
+	}
```

Comment:
```suggestion
	if c, err := semver.NewConstraint(constraint); err != nil || c == nil {
		return fmt.Errorf("error parsing constraint: %w", err)
	}
```

Same for below, this is a good syntax to simplify code.

Reply:
i need the c variable to be out of the scope of the if statement to use it with isValid i think

Reply:
I think in this case u can access it in an else statement.